File: src/components/TweetShareCard.js
import React from 'react';

/**
 * Tweet-style card used only for X sharing.
 * Rendered off-screen and converted to an image.
 */
const TweetShareCard = React.forwardRef(function TweetShareCard(props, ref) {
  const {
    displayName = 'Detea User',
    username = 'detea_user',
    text,
    imageUrl,
    profileImageUrl,
    width = 1080,
    height, // optional; if omitted we derive from width using 7:10 aspect ratio
  } = props;

  // Normalize numeric width/height and enforce width : height = 7 : 10 when height is not provided
  const numericWidth = typeof width === 'number' ? width : 1080;
  const numericHeight =
    typeof height === 'number' ? height : Math.round((numericWidth * 10) / 7);

  return (
    <div
      ref={ref}
      style={{
        width: numericWidth,
        minHeight: numericHeight,
        height: 'auto',
        borderRadius: 24,
        border: '1px solid #E5E7EB',
        backgroundColor: '#FFFFFF',
        padding: 20,
        boxSizing: 'border-box',
        fontFamily:
          '-apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Helvetica, Arial, sans-serif',
        display: 'flex',
        flexDirection: 'column',
      }}
    >
        {/* Header */}
        <div
          style={{
            display: 'flex',
            alignItems: 'center',
            gap: 12,
          }}
        >
          <div
            style={{
              width: 48,
              height: 48,
              borderRadius: '50%',
              overflow: 'hidden',
              backgroundColor: '#EFF3F4',
              flexShrink: 0,
              display: 'flex',
              alignItems: 'center',
              justifyContent: 'center',
            }}
          >
            {profileImageUrl ? (
              <img
                src={profileImageUrl}
                alt={displayName}
                style={{ width: '100%', height: '100%', objectFit: 'cover' }}
              />
            ) : (
              <span
                style={{
                  fontSize: 18,
                  fontWeight: 700,
                  color: '#0F1419',
                }}
              >
                {displayName.charAt(0).toUpperCase()}
              </span>
            )}
          </div>
          <div
            style={{
              display: 'flex',
              flexDirection: 'column',
              minWidth: 0,
            }}
          >
            <span
              style={{
                display: 'flex',
                alignItems: 'center',
                gap: 6,
                minWidth: 0,
                fontSize: 18,
                fontWeight: 700,
                color: '#0F1419',
              }}
            >
              <span style={{ minWidth: 0, overflow: 'hidden', textOverflow: 'ellipsis', whiteSpace: 'nowrap' }}>
                {displayName}
              </span>
              <img
                src={`${process.env.PUBLIC_URL || ''}/DEITECIrc.webp`}
                alt=""
                width={18}
                height={18}
                style={{
                  width: 18,
                  height: 18,
                  flexShrink: 0,
                  objectFit: 'contain',
                  display: 'block',
                }}
              />
            </span>
            <span
              style={{
                marginTop: 2,
                fontSize: 14,
                color: '#536471',
              }}
            >
              @{username}
            </span>
          </div>
        </div>

        {/* Tweet text — no line clamp so full text is visible */}
        <div
          style={{
            fontSize: 16,
            lineHeight: 1.45,
            color: '#0F1419',
            marginTop: 12,
            marginBottom: 16,
            whiteSpace: 'normal',
            wordBreak: 'break-word',
          }}
        >
          {text}
        </div>

        {/* Image */}
        {imageUrl && (
          <div
            style={{
              width: '100%',
              borderRadius: 16,
              overflow: 'hidden',
            }}
          >
            <img
              src={imageUrl}
              alt=""
              style={{
                width: '100%',
                height: 'auto',
                objectFit: 'cover',
                display: 'block',
              }}
            />
          </div>
        )}
    </div>
  );
});

export default TweetShareCard;

File: flutter_app/lib/components/tweet_share_card.dart
import 'package:flutter/material.dart';

import '../utils/hub_carousel_ai_image.dart';

const _kSociteaLogo = 'assets/images/DEITECIrc-192.webp';

/// Tweet-style card used only for X sharing. Rendered off-screen and converted to an image.
class TweetShareCard extends StatelessWidget {
  const TweetShareCard({
    super.key,
    required this.text,
    this.displayName = 'Socitea User',
    this.username = 'socitea_user',
    this.imageUrl,
    this.profileImageUrl,
    this.width = 1080,
    this.height,
  });

  final String displayName;
  final String username;
  final String text;
  final String? imageUrl;
  final String? profileImageUrl;
  final double width;
  final double? height;

  @override
  Widget build(BuildContext context) {
    final h = height ?? (width * 10 / 7);
    return Container(
      width: width,
      constraints: BoxConstraints(minHeight: h),
      padding: const EdgeInsets.all(20),
      decoration: BoxDecoration(
        color: Colors.white,
        borderRadius: BorderRadius.circular(24),
        border: Border.all(color: const Color(0xFFE5E7EB)),
      ),
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.start,
        mainAxisSize: MainAxisSize.min,
        children: [
          Row(
            crossAxisAlignment: CrossAxisAlignment.start,
            children: [
              ClipOval(
                child: Container(
                  width: 48,
                  height: 48,
                  color: const Color(0xFFEFF3F4),
                  child: profileImageUrl != null
                      ? Image.network(profileImageUrl!, fit: BoxFit.cover, width: 48, height: 48)
                      : Center(
                          child: Text(
                            displayName.isNotEmpty ? displayName[0].toUpperCase() : '?',
                            style: const TextStyle(fontSize: 18, fontWeight: FontWeight.w700, color: Color(0xFF0F1419)),
                          ),
                        ),
                ),
              ),
              const SizedBox(width: 12),
              Expanded(
                child: Column(
                  crossAxisAlignment: CrossAxisAlignment.start,
                  children: [
                    Row(
                      children: [
                        Flexible(
                          child: Text(
                            displayName,
                            overflow: TextOverflow.ellipsis,
                            style: const TextStyle(
                              fontSize: 18,
                              fontWeight: FontWeight.w700,
                              color: Color(0xFF0F1419),
                            ),
                          ),
                        ),
                        const SizedBox(width: 6),
                        ClipOval(
                          child: Image.asset(
                            _kSociteaLogo,
                            width: 18,
                            height: 18,
                            fit: BoxFit.cover,
                            errorBuilder: (_, __, ___) => const SizedBox(width: 18, height: 18),
                          ),
                        ),
                      ],
                    ),
                    Text(
                      '@$username',
                      style: const TextStyle(fontSize: 14, color: Color(0xFF536471)),
                    ),
                  ],
                ),
              ),
            ],
          ),
          const SizedBox(height: 12),
          Text(
            text,
            style: const TextStyle(fontSize: 16, height: 1.45, color: Color(0xFF0F1419)),
          ),
          if (imageUrl != null && imageUrl!.isNotEmpty) ...[
            const SizedBox(height: 16),
            ClipRRect(
              borderRadius: BorderRadius.circular(16),
              child: SizedBox(
                width: width - 40,
                child: HubCarouselHeroImage(
                  imageUrl: imageUrl,
                  fit: BoxFit.cover,
                ),
              ),
            ),
          ],
        ],
      ),
    );
  }
}
